dask_expr/_expr.py

```diff
@@ -2454,7 +2489,7 @@ def non_blockwise_ancestors(expr):
         e = stack.pop()
         if isinstance(e, IO):
             yield e
-        elif isinstance(e, Blockwise):
+        elif isinstance(e, BlockwiseOverlapping):
```

Comment:
If this is the only application, then this seems lighter weight to me

```suggestion
        elif isinstance(e, (Blockwise, MapOverlap)):
```